### DISCUSSION TITLE
ShutdownHookPlugin questions

### DISCUSSION THREAD

**Discussion by Bykiev** [quartznet/quartznet#1636] [Q&A] [answered]
Hello @lahma !

I've some questions about ShutdownHookPlugin:
1. Will it be working under IIS (ASP .NET WebApi app, OWIN)
2. Will it be waiting for jobs to complete?
Upd: second question is not a question anymore according to [this ](https://github.com/quartznet/quartznet/blob/90a6bd13b1179277299b032644d4759cbed3e4d9/src/Quartz.Plugins/Plugin/Management/ShutdownHookPlugin.cs#L75)code it's based on `CleanShutdown ` parameter value and it's default value is `true`

Upd2: it seems it won't be working under IIS since ProcessExit is not firing. So, under IIS this plugin is useless and you should manually call shutdown.

**Comment by lahma** [comment, marked as answer]
Like you've already found out, it's probably safer to hook your own code for OWIN callback hooks. I've only work with NET core hosting recently so don't have the answers.
